app/models/miq_group.rb

```diff
@@ -4,7 +4,8 @@ class MiqGroup < ApplicationRecord
   TENANT_GROUP = "tenant"
 
   belongs_to :tenant
-  belongs_to :miq_user_role
+  has_one    :entitlement
+  has_one    :miq_user_role, :through => :entitlement
```

Comment:
I'd assume so. In the future going through to snatch roles from this model is probably a responsibility concern.